core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -675,19 +676,27 @@ public FileInfo getFileInfo(long fileId)
   public FileInfo getFileInfo(AlluxioURI path, GetStatusOptions options)
       throws FileDoesNotExistException, InvalidPathException, AccessControlException {
     Metrics.GET_FILE_INFO_OPS.inc();
-
     try (JournalContext journalContext = createJournalContext();
+         AuditContext auditContext = createAuditContext();
          LockedInodePath inodePath = mInodeTree.lockInodePath(path, InodeTree.LockMode.READ)) {
-      mPermissionChecker.checkPermission(Mode.Bits.READ, inodePath);
+      checkAndAppendAuditLog("getFileInfo", path.toString(), null, auditContext);
+      try {
+        mPermissionChecker.checkPermission(Mode.Bits.READ, inodePath);
```

Comment:
possible to pass `auditContext` to `checkPermission` so let it update the context and whether it is allowed? If that is possible, we could avoid updating the result and perm in multiple return branches in each method.